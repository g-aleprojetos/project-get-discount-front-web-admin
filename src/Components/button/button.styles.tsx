import styled from 'styled-components';
import colors from 'resources/colors';

export interface PropsButton
  extends React.ButtonHTMLAttributes<HTMLButtonElement> {
  tipo?: 'fechar' | 'cancelar' | 'confirmar' | 'ok' | 'submit';
  cor?: string;
}

export const Container = styled.div`
  margin: 15px;
`;

export const ContainerButton = styled.button<PropsButton>`
  min-width: 200px;
  background-color: ${props =>
    props.cor ||
    (props.tipo === 'cancelar'
      ? colors.red
      : props.tipo === 'confirmar'
      ? colors.caribbeanGreen
      : colors.blue)};
  border-radius: ${props => (props.tipo === 'fechar' ? 'none' : '8px')};
  border: none;
  padding: ${props => (props.tipo === 'fechar' ? 'none' : '8px 60px')};
  cursor: pointer;
`;
