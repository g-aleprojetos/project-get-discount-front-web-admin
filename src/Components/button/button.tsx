import React from 'react';
import * as S from './button.styles';

type Props = S.PropsButton & {
  testId?: string;
  children?: React.ReactNode;
  disabled?: boolean;
  aoPressionar?: () => void;
};

export const Button = (props: Props) => {
  const {testId, disabled, tipo, children, aoPressionar} = props;

  return (
    <S.Container data-testid={testId ?? 'bnt-pardao'}>
      <S.ContainerButton
        data-testid={`${testId ? testId + '-button' : 'bnt-pardao-button'}`}
        tipo={tipo}
        type={tipo === 'submit' ? 'submit' : 'button'}
        disabled={disabled}
        onClick={aoPressionar}>
        {children}
      </S.ContainerButton>
    </S.Container>
  );
};
