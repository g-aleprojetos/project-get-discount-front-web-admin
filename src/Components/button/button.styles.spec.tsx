import React from 'react';
import renderer from 'react-test-renderer';
import '@testing-library/jest-dom';
import 'jest-styled-components';
import * as S from './button.styles';

describe('Button.styles', () => {
  test('Container DEVE ser igual ao snapshot', () => {
    const container = renderer.create(<S.Container />).toJSON();

    expect(container).toMatchSnapshot();
  });

  test('ContainerButton DEVE ser igual ao snapshot', () => {
    const container = renderer.create(<S.ContainerButton />).toJSON();

    expect(container).toMatchSnapshot();
  });

  test('ContainerButton DEVE ser igual ao snapshot QUANDO o tipo for "fechar"', () => {
    const container = renderer
      .create(<S.ContainerButton tipo="fechar" />)
      .toJSON();

    expect(container).toMatchSnapshot();
  });

  test('ContainerButton DEVE ser igual ao snapshot QUANDO o tipo for "cancelar"', () => {
    const container = renderer
      .create(<S.ContainerButton tipo="cancelar" />)
      .toJSON();

    expect(container).toMatchSnapshot();
  });

  test('ContainerButton DEVE ser igual ao snapshot QUANDO o tipo for "confirmar"', () => {
    const container = renderer
      .create(<S.ContainerButton tipo="confirmar" />)
      .toJSON();

    expect(container).toMatchSnapshot();
  });

  test('ContainerButton DEVE ser igual ao snapshot QUANDO o tipo for "ok"', () => {
    const container = renderer.create(<S.ContainerButton tipo="ok" />).toJSON();

    expect(container).toMatchSnapshot();
  });

  test('ContainerButton DEVE ser igual ao snapshot QUANDO o tipo for "submit"', () => {
    const container = renderer
      .create(<S.ContainerButton tipo="submit" />)
      .toJSON();

    expect(container).toMatchSnapshot();
  });
});
