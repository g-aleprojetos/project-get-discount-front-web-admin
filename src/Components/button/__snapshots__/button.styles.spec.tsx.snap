// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button.styles Container DEVE ser igual ao snapshot 1`] = `
.c0 {
  margin: 15px;
}

<div
  className="c0"
/>
`;

exports[`Button.styles ContainerButton DEVE ser igual ao snapshot 1`] = `
.c0 {
  background-color: #3D9DB3;
  border-radius: 8px;
  border-style: none;
  padding: 8px 60px;
  cursor: pointer;
}

<button
  className="c0"
/>
`;

exports[`Button.styles ContainerButton DEVE ser igual ao snapshot QUANDO o tipo for "cancelar" 1`] = `
.c0 {
  background-color: #545d6a;
  border-radius: 8px;
  border-style: none;
  padding: 8px 60px;
  cursor: pointer;
}

<button
  className="c0"
/>
`;

exports[`Button.styles ContainerButton DEVE ser igual ao snapshot QUANDO o tipo for "confirmar" 1`] = `
.c0 {
  background-color: #545d6a;
  border-radius: 8px;
  border-style: none;
  padding: 8px 60px;
  cursor: pointer;
}

<button
  className="c0"
/>
`;

exports[`Button.styles ContainerButton DEVE ser igual ao snapshot QUANDO o tipo for "fechar" 1`] = `
.c0 {
  background-color: transparent;
  border-radius: none;
  border-style: none;
  padding: none;
  cursor: pointer;
}

<button
  className="c0"
/>
`;

exports[`Button.styles ContainerButton DEVE ser igual ao snapshot QUANDO o tipo for "ok" 1`] = `
.c0 {
  background-color: #545d6a;
  border-radius: 8px;
  border-style: none;
  padding: 8px 60px;
  cursor: pointer;
}

<button
  className="c0"
/>
`;

exports[`Button.styles ContainerButton DEVE ser igual ao snapshot QUANDO o tipo for "submit" 1`] = `
.c0 {
  background-color: #3D9DB3;
  border-radius: 8px;
  border-style: none;
  padding: 8px 60px;
  cursor: pointer;
}

<button
  className="c0"
/>
`;
